core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2004,7 +2018,7 @@ class Log(@volatile var dir: File,
     maybeHandleIOException(s"Error while deleting log for $topicPartition in dir ${dir.getParent}") {
       lock synchronized {
         checkIfMemoryMappedBufferClosed()
-        removeLogMetrics()
+        close()
```

Comment:
I think we only have one caller of `delete` which is already in a `try` block. Maybe we can let it call `close()` explicitly in the `finally`?

Reply:
reverted.